go/consensus/cometbft/full/archive.go

```diff
@@ -236,3 +249,33 @@ func NewArchive(
 
 	return srv, srv.initialize()
 }
+
+// serviceClientWorker handles command dispatching.
+func (srv *archiveService) serviceClientWorker(ctx context.Context, svc api.ServiceClient) {
+	defer srv.serviceClientsWg.Done()
+
+	sd := svc.ServiceDescriptor()
+	if sd == nil {
+		// Some services don't actually need a worker.
+		return
+	}
+
+	logger := srv.Logger.With("service", sd.Name())
+	logger.Info("starting command dispatcher")
+
+	// Service client event loop.
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case cmd := <-sd.Commands():
```

Comment:
Most apps don't emit commands, so one could stop the routine immediately. 
```go
if sd.Commands() == nil {
    return
```

Reply:
Updated